nrfutil/device/logLibVersions.ts

```diff
@@ -75,24 +77,37 @@ export default async (moduleVersion: ModuleVersion) => {
         const dependencies = moduleVersion.dependencies;
 
         log('nrfutil-device', moduleVersion.version);
-        log('nrf-device-lib', resolveModuleVersion('nrfdl', dependencies));
-        log('nrfjprog DLL', resolveModuleVersion('jprog', dependencies));
-        log('JLink', resolveModuleVersion('JlinkARM', dependencies));
+        log('nrf-device-lib', findDependency('nrfdl', dependencies));
+        log('nrfjprog DLL', findDependency('jprog', dependencies));
+        log('JLink', findDependency('JlinkARM', dependencies));
 
-        const jlinkVersion = resolveModuleVersion('JlinkARM', dependencies);
+        const jlinkCompatibility = getJlinkCompatibility(moduleVersion);
 
-        if (jlinkVersion) {
-            const result = getExpectedVersion(jlinkVersion);
-            if (!result.isExpectedVersion) {
+        switch (jlinkCompatibility.kind) {
+            case 'No J-Link installed':
                 logger?.warn(
-                    `Installed JLink version does not match the expected version (${result.expectedVersion})`
+                    `Segger J-Link is not installed. ` +
+                        `Install at least version ${jlinkCompatibility.requiredJlink} ` +
+                        `from https://www.segger.com/downloads/jlink`
                 );
-            }
-        } else {
-            logger?.warn(
-                `JLink is not installed. Please install JLink from: https://www.segger.com/downloads/jlink`
-            );
+                break;
+            case 'Outdated J-Link':
+                logger?.warn(
+                    `Outdated Segger J-Link. Your version of Segger J-Link (${jlinkCompatibility.actualJlink}) ` +
+                        `is older than the one this app was tested with (${jlinkCompatibility.requiredJlink}). ` +
+                        `Install a newer version from https://www.segger.com/downloads/jlink`
```

Comment:
```suggestion
                        `Install the newer version from https://www.segger.com/downloads/jlink`
```
To make it clear we want that one specific version the app was tested with?
Michael Jones claimed that newer J-Link versions are backward compatible with the older ones, but the broader systems team pointed out there still might be some internal inconsistencies in nrfutil that prevent using any newer J-Link version.

Reply:
Yes, this remains ambiguous. For now I mostly adopt the assumption that newer versions should be fine until we experience that many people have problems with that.